### PR TITLE
feat: added styling to CourseDateBlock

### PR DESCRIPTION
**Pull request by omerhabib26**
### Description

- Added relative icons for each date type
- Added locked content description
- Improve code structuring
- Improve the time format to understand the remaining time & date better

Issue: [156](https://github.com/openedx/openedx-app-android/issues/156)

### CODE REVIEW

**Comment by sdaitzman** [comment]
@omerhabib26 this is looking great, thank you! I noticed two small design issues, hopefully we can resolve these without too much difficulty.

- The Completed Items collapsible block looks like it has around a 3px border, which should be thinner (1px). I think the thicker border draws a bit too much focus on this screen to that box.
- The gray subtext for verification-locked items matches the current Figma designs, but on reviewing this screenshot I noticed it looks a bit hard to read. I checked the contrast ratio and found it's too low to meet accessibility guidelines, so we should update this subtext to the Secondary Dark color in light mode. I've updated the designs in Figma to reflect this change.
A screenshot of a contrast ratio checker showing a contrast ratio of 2.49 for the existing labels, which does not pass the WCAG accessibility guidelines

File: course/src/main/java/org/openedx/course/presentation/dates/CourseDatesFragment.kt [hunk 1]
```diff
@@ -517,13 +535,24 @@ private fun CourseDateItem(
...
+                .padding(end = 4.dp)
                 .clickable(enabled = dateBlock.blockId.isNotEmpty() && dateBlock.learnerHasAccess,
                     onClick = { onItemClick(dateBlock.blockId) })
         ) {
+            dateBlock.dateType.drawableResId?.let { icon ->
+                Icon(
+                    modifier = Modifier
+                        .padding(end = 4.dp)
+                        .align(Alignment.CenterVertically),
+                    painter = painterResource(id = if (dateBlock.learnerHasAccess.not()) coreR.drawable.core_ic_lock else icon),
```
Comment (farhan-arshad-dev) [inline course/src/main/java/org/openedx/course/presentation/dates/CourseDatesFragment.kt:547]: not sure but I think there should be use of `VERIFIED_UPGRADE_DEADLINE` and `VERIFICATION_DEADLINE_DATE`.

File: core/src/main/java/org/openedx/core/data/model/DateType.kt [hunk 2]
```diff
@@ -1,31 +1,32 @@
...
-    CERTIFICATE_AVAILABLE_DATE,
+    CERTIFICATE_AVAILABLE_DATE(R.drawable.core_ic_certificate),
 
     @SerializedName("verified-upgrade-deadline")
-    VERIFIED_UPGRADE_DEADLINE,
+    VERIFIED_UPGRADE_DEADLINE(R.drawable.core_ic_start_end),
 
     @SerializedName("verification-deadline-date")
-    VERIFICATION_DEADLINE_DATE,
+    VERIFICATION_DEADLINE_DATE(R.drawable.core_ic_start_end),
```
[the remark this answers is not in the document]
Comment (farhan-arshad-dev) [reply core/src/main/java/org/openedx/core/data/model/DateType.kt:29]: Also, no use case is defined for these types.

File: course/src/main/java/org/openedx/course/presentation/dates/CourseDatesFragment.kt [hunk 3]
```diff
@@ -537,10 +566,10 @@ private fun CourseDateItem(
                     color = MaterialTheme.appColors.textDark,
                     letterSpacing = 0.15.sp,
                 ),
-                maxLines = 2,
+                maxLines = 1,
                 overflow = TextOverflow.Ellipsis,
             )
-            Spacer(modifier = Modifier.width(16.dp))
+            Spacer(modifier = Modifier.width(7.dp))
```
Comment (farhan-arshad-dev) [inline course/src/main/java/org/openedx/course/presentation/dates/CourseDatesFragment.kt:572]: Not getting the effect of this spacer for me.

File: course/src/main/java/org/openedx/course/presentation/dates/CourseDatesFragment.kt [hunk 4]
```diff
@@ -593,66 +637,70 @@ private val mockedResponse: LinkedHashMap<DatesSection, List<CourseDateBlock>> =
         Pair(
             DatesSection.COMPLETED, listOf(
                 CourseDateBlock(
-                    title = "Homework 1: ABCD",
+                    title = "Homework 1: A B C D",
```
Comment (farhan-arshad-dev) [inline course/src/main/java/org/openedx/course/presentation/dates/CourseDatesFragment.kt:640]: no need to update mock date titles, can be reverted.

**Review by farhan-arshad-dev — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/openedx/core/data/model/DateType.kt [hunk 5]
```diff
@@ -1,31 +1,32 @@
 package org.openedx.core.data.model
 
 import com.google.gson.annotations.SerializedName
+import org.openedx.core.R
 
-enum class DateType {
+enum class DateType(val drawableResId: Int? = null) {
     @SerializedName("todays-date")
     TODAY_DATE,
```
[the remark this answers is not in the document]
Comment (omerhabib26) [reply core/src/main/java/org/openedx/core/data/model/DateType.kt:8]: this type is no longer needed

**Review by k1rill — APPROVED** [review]
Approved from my side

**Review by farhan-arshad-dev — APPROVED** [review]

**Review by volodymyr-chekyrta — APPROVED** [review]
🚀

**Pull request merged**